deployment/common/changeset/save_existing.go

```
@@ -29,9 +33,35 @@ func (cfg ExistingContractsConfig) Validate() error {
 		if err := deployment.IsValidChainSelector(ec.ChainSelector); err != nil {
 			return fmt.Errorf("invalid chain selector: %d - %w", ec.ChainSelector, err)
 		}
-		if ec.Address == (common.Address{}) {
+		if ec.Address == "" {
 			return errors.New("address must be set")
 		}
+		family, err := chain_selectors.GetSelectorFamily(ec.ChainSelector)
+		if err != nil {
+			return err
+		}
+		switch family {
+		case chain_selectors.FamilySolana:
+			decoded, err := base58.Decode(ec.Address)
+			if err != nil {
+				return fmt.Errorf("address must be a valid Solana address (i.e. base58 encoded): %w", err)
+			}
+			if len(decoded) != 32 {
+				return fmt.Errorf("address must be a valid Solana address, got %d bytes expected 32", len(decoded))
+			}
+		case chain_selectors.FamilyEVM:
+			// EVM is the default case
+			fallthrough
+		default:
```

Let's not use this fallthrough pattern, the default case should be loud `panic` with "unsupported chain family" so it's obvious that it's missing support rather than doing the wrong thing. We're taking a similar approach in the core node

---

sounds more robust